classes/i18n/Locale.php

```diff
@@ -389,6 +389,72 @@ public function getFormattedDisplayNames(array $filterByLocales = null, array $l
             ->toArray();
     }
 
+    /**
+     * Getting all translation strings needed in Vue.js UI
+     *
+     *
+     * @return  array Key are translation keys and values are translation strings
+     */
+    public function getUITranslationStrings(): array
+    {
+
+        $filePaths = $this->getUITranslationStringsJsonFilePaths();
+
+        $key = __METHOD__ . static::MAX_CACHE_LIFETIME .  $this->getLocale() . array_reduce($filePaths, fn (string $hash, string $path): string => sha1($hash . $path . filemtime($path)), '');       
+        $expiration = DateInterval::createFromDateString(static::MAX_CACHE_LIFETIME);
+        return Cache::remember($key, $expiration, function () use ($filePaths) {
+            $translations = [];
+
+            // iterate over paths looking for json
+            foreach ($filePaths as $filePath) {
+                $keysJsonData = file_get_contents($filePath);
+                $keysArray = json_decode($keysJsonData, true);
+                foreach ($keysArray as $key) {
+                    if(!$translations[$key]) {
+                        $translations[$key] = $this->translate($key, null, [], null);
+                    }
+                }
+            }
+
+            return $translations;
+        });
+    }
+
+    /**
+     * This is used by TemplateManager to ensure its fetching up to date
+     * translations.
+     *
+     * @return  array Key are translation keys and values are translation strings
+     */
+
+    public function getUITranslationStringsCacheHash(): string
+    {
+        $filePaths = $this->getUITranslationStringsJsonFilePaths();
+
+        $key = static::MAX_CACHE_LIFETIME . $this->getLocale() .  array_reduce($filePaths, fn (string $hash, string $path): string => sha1($hash . $path . filemtime($path)), '');       
+        return sha1($key);
+    }
+
+    /**
+     * Helper function that provides array of existing file paths to uiTranslationKeysBackend.json.
+     * 
+     *
+     * @return  array Values are all paths to existing uiTranslationKeysBackend.json files, including plugins.
+     */
+    private function getUITranslationStringsJsonFilePaths(): array
+    {
+        $filePaths = [];
+
+        foreach (array_keys($this->paths) as $folder) {
+            $filePath = join(DIRECTORY_SEPARATOR, array($folder, 'uiTranslationKeysBackend.json'));
```

Comment:
Interesting.. so php automatically abstracts windows/unix differences? In Node.js I always had to be careful to use these helpers so paths are correct on both platofrms.

Reply:
There's nothing special, Windows has been working fine with the `/` probably since forever :)

You just need to be careful when parsing/storing paths, to avoid having `c:\a/b`, which will work fine, but doesn't look very cute 😅